src/server.c

```diff
@@ -3802,6 +3828,8 @@ int processCommand(client *c) {
                 flagTransaction(c);
             }
             clusterRedirectClient(c,n,c->slot,error_code);
+            /* rejected command can't be rerun */
+            c->flags &= ~CLIENT_RERUN_COMMAND;
```

Comment:
i see you've reset it in `rejectCommand`, and here, but shouldn't it be cleared after every command we've executed (successfully)?
actually, rejections (e.g. OOM). and MOVED, just mean that we actually processed the command (replied to it).
so it's actually the job of `commandProcessed` or `resetClient`.

but then again, maybe this whole change isn't needed now (since this can be handled the same way as #11012)